### DISCUSSION TITLE
Is SuperSlicer enabling ramming and related things when using a wipe tower intentional or a bug?

### DISCUSSION THREAD

**Discussion by wlhlm** [supermerill/SuperSlicer#2205] [Q&A] [answered]
Here is my basic setup:
- OS: Linux
- SS version: 2.4.58.0 (Nigthly branch, commit g2c8b18090)

I have a single extruder, single hotend printer and I want to do a multimaterial print with manual filament change (specifically, the [Voron parts logo](https://github.com/VoronDesign/VoronUsers/tree/master/printer_mods/conqueror433/Voron_Parts_Logo)). Inserting a simple M600 into the GCode is not sufficient as I'm switching between different Materials (from ABS to ABS+) and so want to change filament settings (temp, EM, cooling, speed) as well.

In an old issue I found a [quick guide by merill](https://github.com/supermerill/SuperSlicer/issues/731#issuecomment-735396994) on single extruder multimaterial printing and followed it doing the following:
1. add second extruder in printer settings
2. enable "Single Extruder Multi Material" in printer settings
3. modify toolchange GCode in printer settings to the following: `{if layer_num > 0}M600{endif}`

This does pretty much what I want as it just inserts an M600 at the desired color change point and sets up the filament settings correctly without adding anything else to the GCode (like various stuff for a MMU). A test print worked fine, but I found that adding a wipe tower is probably necessary so that the hotend it properly primed as my first test print had a noticeably seem at the color change. Also merill's old comment I linked above recommends using a wipe tower too. All I did then was to enable the wipe tower in the print settings (the remaining wipe tower settings were left unchange on their default values). Unfortunately, after slicing the GCode now contains extra sections around the M600 filament change that look like they are meant for a MMU (GCode section "CP TOOLCHANGE") including stuff like ramming which I don't want. I'm wondering: is the enabling of MMU GCode with the wipe tower intentional, or a bug? If it's intentional, how can I disable generation of such extra GCode?

Here is an example of the GCode added just by enabling the wipe tower:
```
;--------------------
; CP TOOLCHANGE START
; toolchange #1
; material : ABS -> ABS
;--------------------
M220 S100
; CP TOOLCHANGE UNLOAD
;WIDTH:0.600000
G1  X180.750 Y140.550
G1  X229.478   E1.8520
SET_PRESSURE_ADVANCE ADVANCE=0
[...]
G1  Y142.950
G1  X239.250   E5.0000 F1531       ; <-- this looks like the ramming GCode
G1  X181.250   E-5.0000 F1651
G1  X239.250   E5.0000 F1770
G1  X181.250   E-5.0000 F1889
G1  X239.250   E5.0000 F2008
G1  X181.250   E-5.0000 F2128
G1  X239.250   E5.0000 F2247
G1  X181.250   E-5.0000 F2366
G1 E2.0000 F2000
G1  Y142.900  F2400
G4 S0

M600

G1  X181.250 Y142.900
M221 S92.5 ; extrusion multiplier

SET_PRESSURE_ADVANCE ADVANCE=0.03
[...]
```

Sources:
- [SS project](https://github.com/supermerill/SuperSlicer/files/7836016/mm-cube.3mf.gz)
- [MM print no wipe tower (correct)](https://github.com/supermerill/SuperSlicer/files/7836017/mm-cube-no-wipe.gcode.gz)
- [MM print with wipe tower (incorrect)](https://github.com/supermerill/SuperSlicer/files/7836018/mm-cube-wipe.gcode.gz)

**Comment by supermerill** [comment, marked as answer]
I think so. It can be useful to reduce the amount of plastic left in the nozzle. Also help for retracting, as some printer can be a bit finnicky with thigh extruder gears.
You can disable it if you don't like it.

**Reply by wlhlm** [reply]
Thank you for your response. I initially wasn't able to find the setting to disable the undesired behavior (that's what prompted me to create this thread). But I think I have now found the correct combination of settings that allow me to disable it.

**Reply by supermerill** [reply]
filament -> multimaterial -> toolchange parameters -> ramming (click on it) -> set total ramming time to 0s
